apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -115,39 +344,107 @@ def get_partitions(
         return partitions
 
 
+class TileDBLoader:
+    def __init__(
+        self,
+        uri: str,
+        config: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        from langchain_community.document_loaders.parsers.generic import (
+            MimeTypeBasedParser,
+        )
+        from langchain_community.document_loaders.parsers.html import BS4HTMLParser
+        from langchain_community.document_loaders.parsers.msword import MsWordParser
+        from langchain_community.document_loaders.parsers.pdf import PyMuPDFParser
+        from langchain_community.document_loaders.parsers.txt import TextParser
+
+        self.uri = uri
+        self.config = config
+        self.parser = MimeTypeBasedParser(
+            handlers={
+                "application/pdf": PyMuPDFParser(),
+                "text/plain": TextParser(),
+                "text/html": BS4HTMLParser(),
+                "application/msword": MsWordParser(),
+                "application/vnd.openxmlformats-officedocument.wordprocessingml.document": (
+                    MsWordParser()
+                ),
+            },
+            fallback_parser=None,
+        )
+
+    def load(self) -> List:
+        """Load data into Document objects."""
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Iterator:
+        """A lazy loader for Documents."""
+        import mimetypes
+
+        from langchain_community.document_loaders.blob_loaders import Blob
+
+        vfs = tiledb.VFS(config=self.config)
+        if tiledb.array_exists(self.uri):
+            with tiledb.open(self.uri, "r") as a:
+                mime_type = a.meta.get("mime_type", None)
+            f = tiledb.filestore.Filestore(self.uri)
+        else:
+            mime_type = mimetypes.guess_type(self.uri)[0]
+            f = vfs.open(self.uri)
+
+        if mime_type.startswith("image/"):
+            from langchain_community.document_loaders import UnstructuredFileIOLoader
+
+            unstructured_loader = UnstructuredFileIOLoader(f)
```

Comment:
Does the file store implement the FileIO functions? Can this just be passed to the UnstructuredFileLoader?

Reply:
Tested this with Filestore and it works correctly. UnstructuredFileLoader requires a local file in the filesystem.

Reply:
Sorry that was a typo I meant FileIOLoader. Since the FileStore doesn't fully implemented the python FileIO spec yet was wondering if it was enough. Glad to hear it is!